dali/test/python/test_pipeline.py

```diff
@@ -1091,60 +1091,67 @@ def iter_setup(self):
+                                         num_shards = num_shards,
                                          stick_to_shard = True,
-                                         skip_cached_images = skip_cached_images, 
+                                         skip_cached_images = skip_cached_images,
                                          prefetch_queue_depth = 1)
         elif reader_type == "Caffe2Reader":
-            self.input = ops.Caffe2Reader(path = "/data/imagenet/train-c2lmdb-480", 
-                                          shard_id = 0, 
-                                          num_shards = num_shards, 
+            self.input = ops.Caffe2Reader(path = "/data/imagenet/train-c2lmdb-480",
+                                          shard_id = 0,
+                                          num_shards = num_shards,
                                           stick_to_shard = True,
-                                          skip_cached_images = skip_cached_images, 
+                                          skip_cached_images = skip_cached_images,
                                           prefetch_queue_depth = 1)
         elif reader_type == "FileReader":
             self.input = ops.FileReader(file_root = "/data/imagenet/train-jpeg",
-                                        shard_id = 0, 
-                                        num_shards = num_shards, 
+                                        shard_id = 0,
+                                        num_shards = num_shards,
                                         stick_to_shard = True,
-                                        skip_cached_images = skip_cached_images, 
+                                        skip_cached_images = skip_cached_images,
                                         prefetch_queue_depth = 1)
 
         elif reader_type == "TFRecordReader":
             tfrecord = sorted(glob.glob("/data/imagenet/train-val-tfrecord-480/train-*"))
             tfrecord_idx = sorted(glob.glob("/data/imagenet/train-val-tfrecord-480.idx/train-*"))
             self.input = ops.TFRecordReader(path = tfrecord,
                                             index_path = tfrecord_idx,
-                                            shard_id = 0, 
-                                            num_shards = num_shards, 
+                                            shard_id = 0,
+                                            num_shards = num_shards,
                                             stick_to_shard = True,
-                                            skip_cached_images = skip_cached_images, 
+                                            skip_cached_images = skip_cached_images,
                                             features = {"image/encoded" : tfrec.FixedLenFeature((), tfrec.string, ""),
                                                         "image/class/label": tfrec.FixedLenFeature([1], tfrec.int64,  -1)})
 
-        if is_cached:
+        if is_cached or is_cached_batch_copy:
```

Comment:
Shouldnt here just be `if is_cached_batch_copy` and next condition `   if is_cached` ?

I think `is_cached_batch_copy` should just override `is_cached`, so you dont have to care about ` is_cached_batch_copy == False` in next condition

Reply:
Here I'd limit the condition to:
```
if is_cached:
```